collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -0,0 +1,199 @@
+
+import (
+	"context"
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/golang-collections/go-datastructures/queue"
+	"github.com/open-telemetry/opentelemetry-lambda/collector/internal/telemetryapi"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	semconv "go.opentelemetry.io/collector/semconv/v1.5.0"
+	"go.uber.org/zap"
+)
+
+const defaultListenerPort = "4325"
+const initialQueueSize = 5
+
+type telemetryAPIReceiver struct {
+	httpServer            *http.Server
+	logger                *zap.Logger
+	queue                 *queue.Queue // queue is a synchronous queue and is used to put the received log events to be dispatched later
+	nextConsumer          consumer.Traces
+	lastPlatformStartTime string
+	lastPlatformEndTime   string
+	extensionID           string
+}
+
+func (r *telemetryAPIReceiver) Start(ctx context.Context, host component.Host) error {
+	address := listenOnAddress()
+	r.logger.Info("Listening for requests", zap.String("address", address))
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", r.httpHandler)
+	r.httpServer = &http.Server{Addr: address, Handler: mux}
+	go func() {
+		_ = r.httpServer.ListenAndServe()
+	}()
+
+	telemetryClient := telemetryapi.NewClient(r.logger)
+	_, err := telemetryClient.Subscribe(ctx, r.extensionID, fmt.Sprintf("http://%s/", address))
+	if err != nil {
+		r.logger.Info("Listening for requests", zap.String("address", address), zap.String("extensionID", r.extensionID))
+		return err
+	}
+	return nil
+}
+
+func (r *telemetryAPIReceiver) Shutdown(ctx context.Context) error {
+	return nil
+}
+
+func newSpanID() pcommon.SpanID {
+	var rngSeed int64
+	_ = binary.Read(crand.Reader, binary.LittleEndian, &rngSeed)
+	randSource := rand.New(rand.NewSource(rngSeed))
+	sid := pcommon.SpanID{}
+	_, _ = randSource.Read(sid[:])
+	return sid
+}
+
+// httpHandler handles the requests coming from the Telemetry API.
+// Everytime Telemetry API sends events, this function will read them from the response body
+// and put into a synchronous queue to be dispatched later.
+// Logging or printing besides the error cases below is not recommended if you have subscribed to
+// receive extension logs. Otherwise, logging here will cause Telemetry API to send new logs for
+// the printed lines which may create an infinite loop.
+func (r *telemetryAPIReceiver) httpHandler(w http.ResponseWriter, req *http.Request) {
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		r.logger.Error("error reading body", zap.Error(err))
+		return
+	}
+
+	var slice []event
+	if err := json.Unmarshal(body, &slice); err != nil {
+		r.logger.Error("error unmarshalling body", zap.Error(err))
+		return
+	}
+
+	for _, el := range slice {
+		r.logger.Info(fmt.Sprintf("Event: %s", el.Type), zap.Any("event", el))
```

Comment:
Is this maybe a bit verbose for `Info`?  I'd put event details as `Debug`.

Reply:
Updated.